### ISSUE TITLE
[Bug]: Auckland City Council stopped working around update to HA 2024.4.2 or 1.47.0

### ISSUE THREAD

**Issue by xbmcnut** [mampfes/hacs_waste_collection_schedule#1962]
### I Have A Problem With:

A specific source

### What's Your Problem

All sensors for `aucklandcouncil_govt_nz` now show unavailable. Seems to have stopped working around either my recent update to 1.47.0 or HA 2024.4.2. Configuration has not changed at all. Tried downgrading to 1.45.0 and 1.46.0, still the same. Reinstalled 1.47.0 and still no go.

![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/1492137/44bc1f6d-4a70-46ea-be25-7db66dcae72b)

### Source (if relevant)

aucklandcouncil_govt_nz

### Logs

```Shell
no relevant logs
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: aucklandcouncil_govt_nz
      args:
        area_number: redacted
        # from https://www.aucklandcouncil.govt.nz/rubbish-recycling/rubbish-recycling-collections/Pages/rubbish-recycling-collection-days.aspx
      customize:
        - type: rubbish
          alias: Rubbish
          icon: mdi:trash-can
        - type: recycle
          alias: Recycling
          icon: mdi:recycle
      calendar_title: Auckland Council Rubbish & Recycling Collections
  day_switch_time: "17:00"
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [ ] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by sahne2000** [comment]
Your are not alone. Same issue. I had a quick look yesterday and from what I can see Auckland council website has slightly changed. I could not finish it actual work task took over. Hopefully today or someone else has time to adjust.

**Comment by BambamNZ** [comment]
I've looked into this as well, as my implementation stopped working - Status change to Unknown
Schedule was working post upgrade to 2024.4.2 & 1.47.0 based on my looking back at logs, also tested with downgrades - no change. I tried a random address from a other NZ council and that worked.
Leads me to believe it is a change at Auckland council end which results in data not being pulled into HA / Waste Collection Schedule

**Comment by nicwise** [comment]
For some reason, I think it's cos the council changed their site somehow. The code made no sense vrs the actual site HTML

I rewrite it using Beuitiful Soup, which is already part of this plugin

https://github.com/mampfes/hacs_waste_collection_schedule/pull/1968

**Comment by sahne2000** [comment]
@nicwise  You beat me to it. I was too slow. All working. For those who cant wait just replace the source file manually.

**Comment by xbmcnut** [comment]
> @nicwise You beat me to it. I was too slow. All working. For those who cant wait just replace the source file manually.

Thanks for the fast action, much appreciated. Dumb question but where is the new source file I can try?

**Comment by xbmcnut** [comment]
Ignore my last question, found it [here](https://github.com/mampfes/hacs_waste_collection_schedule/blob/427293018cc050cb2726ddd7e9e7c52c93b19259/custom_components/waste_collection_schedule/waste_collection_schedule/source/aucklandcouncil_govt_nz.py). Downloaded raw file and replaced the file `aucklandcouncil_govt_nz.py` in this location \`\config\custom_components\waste_collection_schedule\waste_collection_schedule\source` and it's now working. Thank you!

![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/1492137/57445357-ee1c-446d-b334-713b49345f8a)

![image](https://github.com/mampfes/hacs_waste_collection_schedule/assets/1492137/c6a839fa-f700-444d-8632-63751177e764)

**Comment by nicwise** [comment]
nice, now I need to go do the same thing.

**Comment by gnt-nz** [comment]
Thank you so much for fixing this.

**Comment by Zetto12** [comment]
Thanks for the quick fix. Looks like it changed a bit, is that possible? The Auckland council sensor is back online but the template sensors I had are not working anymore. Anyone knows why?

#Sensor for waste collection
# next collection
- platform: waste_collection_schedule
  name: next_collection
# next yard waste collection
- platform: waste_collection_schedule
  name: next_rubbish_collection
  types:
    - Rubbish
# next garbage collection
- platform: waste_collection_schedule
  name: next_recycling_collection
  types:
    - Recycling
#Custom sensor to show days until next waste collection
- platform: template
  sensors:
    days_until_collection:
      friendly_name: "Days Until Collection"
      value_template: >-
        {% set sensor_state = states('sensor.next_collection') %}
        {% set days_index = sensor_state.find('day') %}
        {% set days = sensor_state[sensor_state.find('in ') + 3: days_index].strip() %}
        {{ days }}
      unit_of_measurement: "days"
    type_of_waste:
      friendly_name: "Type of Waste"
      value_template: >-
        {% set sensor_state = states('sensor.next_collection') %}
        {% set in_index = sensor_state.find('in') %}
        {% set type_of_waste = sensor_state[:in_index].strip() %}
        {% set replaced_type_of_waste = type_of_waste[:type_of_waste.rfind(',')] + ' and' + type_of_waste[type_of_waste.rfind(',')+1:] %}
        {{ replaced_type_of_waste }}

**Comment by 5ila5** [comment]
As it led to some confusion:

This fix will be part of release 1.48.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
